### PR TITLE
ivi-controller: fix memory leak of background surface remove

### PR DESCRIPTION
**Pull request by khangtb1**
ivi-controller: fix memory leak of background surface remove

### CODE REVIEW

File: weston-ivi-shell/src/ivi-controller.c [hunk 1]
```diff
@@ -1723,36 +1723,45 @@ surface_event_remove(struct wl_listener *listener, void *data)
     struct notification *not, *next;
```
Comment (HarshaMM) [inline weston-ivi-shell/src/ivi-controller.c:1724]: How about refactoring the code to two functions? (names below can be better):
-> remove_bkgnd_surface
-> remove_shell_surface

Implementation does look much cleaner then

**Review by HarshaMM — APPROVED** [review]
`Reviewed-by: Harsha M M <harsha.manjulamallikarjun@in.bosch.com`

**Review by efriedrich — APPROVED** [review]
Reviewed-by: Eugen Friedrich <efriedrich@de.adit-jv.com>

**Comment by khangtb1** [comment]
> @khangtb1 please resolve the merge conflict on you pr

Done. Please check!

**Review by efriedrich — APPROVED** [review]

**Pull request merged**